File: loan_management_system/models/Loan.js
const mongoose = require('mongoose');

const LoanSchema = new mongoose.Schema({
  client: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Client',
    required: true,
  },
  amount: {
    type: Number,
    required: [true, 'El monto del préstamo es obligatorio.'],
  },
  interestRate: {
    type: Number,
    required: [true, 'La tasa de interés es obligatoria.'],
  },
  totalAmount: {
    type: Number,
    required: true,
  },
  installments: {
    type: Number,
    required: [true, 'El número de cuotas es obligatorio.'],
  },
  installmentAmount: {
    type: Number,
    required: true,
  },
  period: {
    type: String,
    enum: ['daily', 'weekly'],
    required: [true, 'El período (diario/semanal) es obligatorio.'],
  },
  issueDate: {
    type: Date,
    default: Date.now,
  },
  status: {
    type: String,
    enum: ['active', 'paid'],
    default: 'active',
  },
});

module.exports = mongoose.model('Loan', LoanSchema);

File: loan_management_system/android_app/app/src/main/java/com/example/loanmanagement/data/model/Loan.kt
package com.example.loanmanagement.data.model

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName

@Entity(
    tableName = "loans",
    foreignKeys = [
        ForeignKey(
            entity = Client::class,
            parentColumns = ["id"],
            childColumns = ["clientId"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class Loan(
    @PrimaryKey
    @SerializedName("_id")
    val id: String,

    @SerializedName("client")
    val clientId: String,

    @SerializedName("amount")
    val amount: Double,

    @SerializedName("interestRate")
    val interestRate: Double,

    @SerializedName("totalAmount")
    val totalAmount: Double,

    @SerializedName("installments")
    val installments: Int,

    @SerializedName("installmentAmount")
    val installmentAmount: Double,

    @SerializedName("period")
    val period: String, // "daily" or "weekly"

    @SerializedName("issueDate")
    val issueDate: String,

    @SerializedName("status")
    val status: String // "active" or "paid"
)
